Added section on enabling fixed egress ips to managing networking topic

bfallonf opened the pull request:
For: https://trello.com/c/iyLkQ8Yu/688-document-semi-automatic-namespace-wide-egress-ip-egress

cc @danwinship

danwinship commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +872,82 @@ is complicated, because nodes have both "external" IP addresses and
+oc patch netnamespace MyProject -p '{"egressIPs": ["192.168.1.100"]}'
+----
++
+The `egressIPs` field can be an array, but still must be set to a single IP address. If setting multiple IPs, the other IPs will be ignored.
```
"is" not "can be"

"The `egressIPs` field is an array, but must be set to a single IP address."

danwinship commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +872,82 @@ is complicated, because nodes have both "external" IP addresses and
+The `egressIPs` field can be an array, but still must be set to a single IP address. If setting multiple IPs, the other IPs will be ignored.
+
+//The following steps can maybe be deleted when tech preview is over when this process if automatic:
+. Manually assign the egress IP to the desired node hosts. Set the `egressIPs` field on the `HostSubnet` object on the node host. Include as many IPs as you want the node host to be set to:
```
"Include as many IPs as you want to assign to that node:"

danwinship commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +872,82 @@ is complicated, because nodes have both "external" IP addresses and
+  '{"egressIPs": ["<IP_address_1>", "<IP_address_2>"]}'
+----
++
+For example, to say that node1 should be set the egress IPs 192.168.1.100, 192.168.1.101, and 192.168.1.102:
```
"that node1 should have the egress IPs" ?

danwinship commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +872,82 @@ is complicated, because nodes have both "external" IP addresses and
+====
+Egress IPs are implemented as additional IP addresses on the primary network
+interface, and must be in the same subnet as the node's primary IP. This might
+require extra configuration when using some cloud or VM solutions.
```
Hm... that kinda makes it sound like it's the "same subnet" part that may require extra configuration, when it's really the "additional IPs" part.

"Allowing additional IP addresses on the primary network interface might require extra configuration when using some cloud or VM solutions." ?

We should try to figure out the details here... whoever provided the notes on vmware/aws/etc for egress-router should be able to help with this. (Doesn't have to happen as part of this PR, we can add more details as we figure them out.)

bfallonf replied:
> We should try to figure out the details here... whoever provided the notes on vmware/aws/etc for egress-router should be able to help with this. (Doesn't have to happen as part of this PR, we can add more details as we figure them out.)

I agree. I can make a followup PR for sure.

bfallonf commented:
Thanks @danwinship . I changed to the few suggestions and added a paragraph in the "Controlling Egress Traffic" section. Any more thoughts, or can I put this to review?

danwinship commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
+[IMPORTANT]
+====
+Enabling fixed IPs for external project traffic is a Technology Preview feature
+only. 
```
not sure if it would be "in character" for our docs, but maybe add a note here about the fact that we plan to add further functionality to this feature in the future.

bfallonf replied:
I think you're right that it's not too in character. It's more worthwhile putting that information in when it's available, I think (and I think that's how we usually do it).

If peer review people disagree, please let me know.

mburke5678 replied:
@bfallonf I think I have seen such references (I am not a fan, but....)
For ex:
`A future release of {product-title} will feature specific support for per-project back up and restore.`

bfallonf commented:
Thanks @danwinship . I'll move forward with this.

@openshift/team-documentation peer review plz?

bmeng commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
+The `egressIPs` field is an array, but must be set to a single IP address. If setting multiple IPs, the other IPs will be ignored.
+
+//The following steps can maybe be deleted when tech preview is over when this process if automatic:
+. Manually assign the egress IP to the desired node hosts. Set the `egressIPs` field on the `HostSubnet` object on the node host. Include as many IPs as you want to assign to that node host:
```
This also has the index `1`?

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -142,6 +142,13 @@ call out to the egress router pod's service
 (`service.project.cluster.domainname.com`) in their application (for example,
 the JDBC connection information) rather than the actual protected service URL.
 
+You can also fix specific IP addresses to projects, ensuring that all outgoing
```
Is "fix" the right word? associate? assign?

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -142,6 +142,13 @@ call out to the egress router pod's service
 the JDBC connection information) rather than the actual protected service URL.
 
+You can also fix specific IP addresses to projects, ensuring that all outgoing
+external connections from the specified project have recognizable origins. This
```
Is `fix` the right word here? Sounds like it is broken! 
Assign?

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
 bypass it.
 
+[[enabling-fixed-ips-for-external-project-traffic]]
+== Enabling Fixed IPs for External Project Traffic 
```
I did a quick Google on Fixed IPs and all of the references came up as `Static IP`. Is that a better term?
We use Static IP in a few places in our docs, but no Fixed IP that I saw in Atom

bfallonf replied:
Hmm I do that these as two separate things, but I'm no expert. @danwinship thoughts?

danwinship replied:
We haven't been very consistent. "fixed", "static", "reserved", "private"...

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
+oc patch netnamespace <project_name> -p '{"egressIPs": ["<IP_address>"]}'
+----
++
+For example, to assign the MyProject project to an IP address of 192.168.1.100:
```
Does `MyProject` need markup? the *MyProject* project??

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
+  '{"egressIPs": ["<IP_address_1>", "<IP_address_2>"]}'
+----
++
+For example, to say that node1 should have the egress IPs 192.168.1.100, 192.168.1.101, and 192.168.1.102:
```
`node1` ??

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
+====
+Egress IPs are implemented as additional IP addresses on the primary network
+interface, and must be in the same subnet as the node's primary IP. Allowing additional IP addresses on the primary network interface might require extra configuration when using some cloud or VM solutions.
+====
```
Extra configuration where? In OpenShift or the cloud or VM solution?

bfallonf replied:
Yes, this is something I talked to @danwinship about. Will require a bigger followup.

mburke5678 commented on admin_guide/managing_networking.adoc:
```diff
@@ -872,6 +879,81 @@ is complicated, because nodes have both "external" IP addresses and
+====
+
+If the above is enabled for a project, all egress traffic from that project will
+be routed to the node hosting that egress IP, then NATted to that IP address. If
```
NATted. Should this be NAT'd? 
(Natted in the urban dictionary is a very different thing! http://www.urbandictionary.com/define.php?term=natted )

bfallonf replied:
Hmm right. Replaced with "connected (using NAT)"

mburke5678 commented:
@bfallonf Couple of comments.

bfallonf commented:
[rev_history]
|xref:../admin_guide/managing_networking.adoc#admin-guide-manage-networking[Managing Networking]
|Added the xref:../admin_guide/managing_networking.adoc#enabling-static-ips-for-external-project-traffic[Enabling Static IPs for External Project Traffic] section.
%

gaurav-nelson commented:
@bfallonf Does this need to go in dedicated branch as well?
